nengo/cache.py

```diff
@@ -365,6 +370,10 @@ def wrap_solver(self, solver_fn):
         func
             Wrapped decoder solver.
         """
+        if self._index is None:
+            warnings.warn("Cannot wrap solver outside of `with cache` block.")
```

Comment:
I think technically you could as long as the call to the solver will be within the `with` block. Also, it would be possible to do the solver call after the `with` block which would be a problem.

Maybe this check should be moved into `cached_solver`?